pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -530,6 +537,13 @@ func (p *PriorityQueue) DeleteNominatedPodIfExists(pod *v1.Pod) {
 	p.lock.Unlock()
 }
 
+// HasUnschedulablePods returns true if any unschedulable pods exist in the SchedulingQueue.
+func (p *PriorityQueue) HasUnschedulablePods() bool {
+	p.lock.Lock()
```

Comment:
There is no write operation here, and p.lock is a `sync.RWMutex`, so I recommend you to use `p.lock.RLock().` 
(FYI:https://golang.org/pkg/sync/#RWMutex)

Reply:
+1. This should be a read lock.